### ISSUE TITLE
[FEATURE REQUEST] Add option to disable LCD Communication Check in secret menu.

### ISSUE THREAD

**Issue by CuriousTimo** [prusa3d/Prusa-Firmware-Buddy#3561]
**Printer type** - [MINI]

**Is your feature request related to an existing problem? Please describe.**
  This feature request is related to issue [3277](https://github.com/prusa3d/Prusa-Firmware-Buddy/issues/3277) where the LCD Screen keeps flashing on the MINI with a clone replacement LCD Screen.

**Describe the expected functionality**
If we make `lcd::communication_check(); ` in [Prusa-Firmware-Buddy/src/gui/guimain.cpp](https://github.com/prusa3d/Prusa-Firmware-Buddy/blob/15d5ae0b317a37b69b8e7a4720711b962c25b57d/src/gui/guimain.cpp#L474) configurable using an option in the secret menu, MINI users using an affected LCD screen, can disable the LCD Communication Check to 'fix' their flashing screen.

**Root Cause for Flashing Screen issue  [3277](https://github.com/prusa3d/Prusa-Firmware-Buddy/issues/3277)**
The root cause for the flashing of the LCD seems to be that these screens do not have IM3 pulled high.
This causes the st7789 to run on half duplex mode
```
SDA:
-When IM3: Low, SPI interface input/output pin
-When IM3: High, SPI interface input pin
```

So to quote @pchrist from issue 3277:

> I finally got hold of a Fysetc display, I hooked up a logic analyzer to the SPI signals between the MCU and the display controller. I found out that a read command to the display is issued every 2 seconds (on the SDA line). On a Prusa display the response is seen on the SDO line. On the Fysetc display the response is seen on the SDA line.
> 
> It looks like the Prusa display is running in full duplex mode and the Fysetc display is running in half duplex mode. As long as nothing is read back from the display this does not matter. Since release 4.6.1 a LCD communication check is executed every 2 seconds. The check will fail when running half duplex mode, resulting in a display reset, seen as a white screen.
> 
> The half/full duplex configuration is done on the display by pulling some configuration pins high or low. These pins are not easily accessible on the Fysetc display.

**Comment by FrankCellini** [comment]
I have same problem... please Prusa fix this display issue by installing an option to disable LCD Communication check.  I only purchased this display because the incremental encoder broke so the tech at Prusa recommended I buy the fysetc display which is larger and easier to read and it came with  incremental encoder on board saving me removing and replacing encoder.

**Comment by hreality** [comment]
Same, I was using a previous community compiled FW to resolve the flashing LCD while waiting for the final release.. only to flash the final and be back to my screen flicker.

**Comment by kayari75** [comment]
I have the same problem too. I use a bigger fysetc display  and it keep flashing on all screens except on the Live Adjust Z menu.

**Comment by hreality** [comment]
REALLY looking forward to this getting resolved.  Absolutely love my mini but that flashing screen is an eyesore.

**Comment by FrankCellini** [comment]
Yes I too am looking forward to getting it fixed… I think Prusa have given up on this?

--

Frank Cellini
Mobile: 0414 385 344
E-mail: ***@***.***
